3rdParty/EndianPortable/include/EndianPortable.h

```diff
@@ -61,25 +61,25 @@
    #define le32toh(x) (x)
   #endif
  #endif
-
 #elif defined __CYGWIN__
  #include <endian.h>
-#elif defined(_WIN32)
-#include <winsock2.h>
+#elif defined(_WIN32) && !defined(_MSC_VER)
+  // MinGW32
+ #include <winsock2.h>
  #if(BYTE_ORDER == LITTLE_ENDIAN)
-  #define htobe16(x) htons(x)
+  #define htobe16(x) __builtin_bswap16(x)
   #define htole16(x) (x)
-  #define be16toh(x) ntohs(x)
+  #define be16toh(x) __builtin_bswap16(x)
   #define le16toh(x) (x)
 
-  #define htobe32(x) htonl(x)
+  #define htobe32(x) __builtin_bswap32(x)
   #define htole32(x) (x)
-  #define be32toh(x) ntohl(x)
+  #define be32toh(x) __builtin_bswap32(x)
   #define le32toh(x) (x)
 
-  #define htobe64(x) htonll(x)
+  #define htobe64(x) __builtin_bswap64(x)
   #define htole64(x) (x)
-  #define be64toh(x) ntohll(x)
+  #define be64toh(x) __builtin_bswap64(x)
```

Comment:
Why is this change needed? Wouldn't it be simpler to keep the previous code? That way we don't need to add a special `#ifedf` for MSVC

Reply:
But htonll isn't defined and is it guarantee that network byte order is always big endian ?

Reply:
If you want to keep the previous format,  the Big Endian on windows will fail as it's using GCC built-in

Reply:
`htonll` isn't defined in which platform? We currently don't use it so I'm not sure it matters...
Anyway I assume this is not needed for this PR, is that correct?
Maybe we can remove this change from this PR and think later if we want to add it or not. What do you think?

Reply:
We don't use but C preprocessor comply that the macro doesn't exist.

Yes not related to this MR, I will drop these changes

Reply:
@seladb done